include/ginkgo/core/matrix/fbcsr.hpp

```diff
@@ -0,0 +1,390 @@
+ * A->apply(alpha, I, beta, B) // B = alpha*A + beta*B
+ * ```
+ * Both the SpGEMM and SpGEAM operation require the input matrices to be sorted
+ * by block-column index, otherwise the algorithms will produce incorrect
+ * results.
+ *
+ * @tparam ValueType  precision of matrix elements
+ * @tparam IndexType  precision of matrix indexes
+ *
+ * @ingroup fbcsr
+ * @ingroup mat_formats
+ * @ingroup LinOp
+ */
+template <typename ValueType = default_precision, typename IndexType = int32>
+class Fbcsr : public EnableLinOp<Fbcsr<ValueType, IndexType>>,
+              public EnableCreateMethod<Fbcsr<ValueType, IndexType>>,
+              public ConvertibleTo<Fbcsr<next_precision<ValueType>, IndexType>>,
+              public ConvertibleTo<Dense<ValueType>>,
+              public ConvertibleTo<Csr<ValueType, IndexType>>,
+              public ConvertibleTo<SparsityCsr<ValueType, IndexType>>,
+              public DiagonalExtractable<ValueType>,
+              public ReadableFromMatrixData<ValueType, IndexType>,
+              public WritableToMatrixData<ValueType, IndexType>,
+              public Transposable,
+              public EnableAbsoluteComputation<
+                  remove_complex<Fbcsr<ValueType, IndexType>>> {
+    friend class EnableCreateMethod<Fbcsr>;
+    friend class EnablePolymorphicObject<Fbcsr, LinOp>;
+    friend class Dense<ValueType>;
+    friend class SparsityCsr<ValueType, IndexType>;
+    friend class FbcsrBuilder<ValueType, IndexType>;
+    friend class Fbcsr<to_complex<ValueType>, IndexType>;
+
+public:
+    using value_type = ValueType;
+    using index_type = IndexType;
+    using transposed_type = Fbcsr<ValueType, IndexType>;
+    using mat_data = matrix_data<ValueType, IndexType>;
+    using absolute_type = remove_complex<Fbcsr>;
+
+
+    void convert_to(Fbcsr<ValueType, IndexType> *result) const override;
+
+    void move_to(Fbcsr<ValueType, IndexType> *result) override;
+
+    friend class Fbcsr<next_precision<ValueType>, IndexType>;
+
+    void convert_to(
+        Fbcsr<next_precision<ValueType>, IndexType> *result) const override;
+
+    void move_to(Fbcsr<next_precision<ValueType>, IndexType> *result) override;
+
+    void convert_to(Dense<ValueType> *other) const override;
+
+    void move_to(Dense<ValueType> *other) override;
+
+    void convert_to(Csr<ValueType, IndexType> *result) const override;
+
+    void move_to(Csr<ValueType, IndexType> *result) override;
+
+    /**
+     * Get the block sparsity pattern in CSR-like format
+     *
+     * Note that the actual non-zero values are never copied;
+     * the result always has a value array of size 1 with the value 1.
+     */
+    void convert_to(SparsityCsr<ValueType, IndexType> *result) const override;
+
+    void move_to(SparsityCsr<ValueType, IndexType> *result) override;
+
+    /** Convert COO data into block CSR
+     *
+     * @warning Unlike Csr::read, here explicit non-zeros are NOT dropped.
+     */
```

Comment:
This is a technical issue but I'm not sure what happens in this case to the documentation, is it appended or does it replace it? If it replaces, it could be worth trying to use the command `@copydoc LinOp::read` instead.

Reply:
I guess you mean `ReadableFromMatrixData::read`. But there are two read functions there, so it might be better to not do that. I'll just improve the current docstring.